DP-38339-Add-unique-ID-to-login-dropdown-component

jarielrrz opened the pull request:
This PR adds a conditional id attribute in button-dropdown.twig, so we can set gtm-loginto on the "Log In To" button for tracking.

jarielrrz committed:
Update button-dropdown.twig

adds a conditional id attribute in button-dropdown.twig, so we can set gtm-loginto on the "Log In To" button for tracking.

jarielrrz committed:
Create DP-38339.yml

adds a change log.

dstorozhuk commented on packages/patternlab/styleguide/source/_patterns/02-molecules/button-dropdown.twig:
```diff
@@ -5,7 +5,8 @@
       'aria-haspopup': 'true',
-      'aria-controls': menuId|default('')
+      'aria-controls': menuId|default(''),
+      'id': button.id|default(null)
```
@jarielrrz  if button.id is empty - does id attribute exist on the element?

jarielrrz committed:
Update button.twig

This ensures that the id attribute is only added to the button if it is explicitly provided.

jarielrrz committed:
Update page-header.twig

Pass the `button` object as-is to `@molecules/button-dropdown.twig` and `@atoms/01-buttons/button.twig`, allowing OpenMass to provide the `id` if needed.

jarielrrz committed:
Revert "Update page-header.twig"

This reverts commit 1861c6146fb60101ea9f78892fcf38ff28825d59.

jarielrrz committed:
Merge branch 'develop' into patternlab/DP-38339-Add-unique-ID-to-login-dropdown-component

jarielrrz committed:
Revert "Update button-dropdown.twig"

This reverts commit 203728e05bf21616ab3bd37ee642c6d091c056cb.

jarielrrz committed:
Merge branch 'patternlab/DP-38339-Add-unique-ID-to-login-dropdown-component' of https://github.com/massgov/mayflower into patternlab/DP-38339-Add-unique-ID-to-login-dropdown-component
